lightning/src/util/enforcing_trait_impls.rs

```diff
@@ -24,19 +24,32 @@
 use std::io::Error;
 use ln::msgs::DecodeError;
 
+/// Initial value for revoked commitment downward counter
+pub const INITIAL_REVOKED_COMMITMENT_NUMBER: u64 = 1 << 48;
```

Comment:
I'm probably missing something obvious about u64s or of a BOLT, but why is the bit shift 48?

Reply:
Revocation secrets are computed using backwards-counting commitment numbers, starting at `(1<<48) - 1`.  The BOLTs keep track of commitment numbers starting at zero.

I'm not sure exactly why LDK keeps track of commitment numbers everywhere using the backwards scheme instead of the forwards scheme.  You can see all the places by searching for `1 << 48`.   Maybe @TheBlueMatt or @ariard know.

Reply:
Sadly we need both the backwards- and forwards-counting versions in various places. The BOLTs didn't specify them in either direction and I initially implemented them backwards counting for...reasons unknown. There's not much reason to change it, though, since we need both either way (IIRC).